libkbfs/root_metadata.go

```diff
@@ -843,23 +391,46 @@ func (md *RootMetadata) swapCachedBlockChanges() {
 	}
 }
 
-// TlfHandleExtensions returns a list of handle extensions associated with the TLf.
-func (md *RootMetadata) TlfHandleExtensions() (extensions []TlfHandleExtension) {
-	if md.ConflictInfo != nil {
-		extensions = append(extensions, *md.ConflictInfo)
-	}
-	if md.FinalizedInfo != nil {
-		extensions = append(extensions, *md.FinalizedInfo)
+// A ConstRootMetadata is a thin wrapper around a
```

Comment:
I wanted to avoid names being too long, but you're right. Went with `ReadOnlyRootMetadata` since that has a more accurate connotation than `Const` (which has some baggage from C++) and clarified comments to just say that you should be careful of aliasing if you store `ReadOnlyRootMetadata` objects.

Also added a `.ReadOnly()` method to `*RootMetadata*`, which helps with the long name problem, and used it everywhere (as a bonus, it also works on `ImmutableRootMetadata`).